### ISSUE TITLE
Delta Report giving error

### ISSUE THREAD

**Issue by Manbearpiet** [microsoft/Microsoft365DSC#1880]
I created a dev tenant to test out M365 DSC. I've created an initial export which contains a lot of data in the initial export. Now I created a group and a MEM device policy, which I wanted to report on using the delta report.

# Shown behavior

Using the following commands I tried to create a delta export:
```powershell
New-M365DSCDeltaReport -Source .\M365TenantConfig.ps1 -Destination .\New.ps1 -OutputPath .\delta.html
Error: Settings
Error: Settings
```

This renders:
![image](https://user-images.githubusercontent.com/43993129/162943605-e989be54-f0d9-4105-81ff-5c0a04b146d6.png)

Another attempt:
```powershell
New-M365DSCDeltaReport -Source .\M365TenantConfig.ps1 -Destination .\New.ps1 -OutputPath .\delta.html -Delta $true -Verbose
VERBOSE: Obtaining Delta between the source and destination configurations
```
![image](https://user-images.githubusercontent.com/43993129/162944186-fa345cad-b888-42a6-b88e-1468d3b9716c.png)

I'm sure the configurations contain seperate info, due to the fact the output contained it:

Part of M365TenantConfig.ps1:

```powershell
[3/137] Extracting [AADGroup]...
    |---[1/8] All Company✅
    |---[2/8] Retail✅
    |---[3/8] Mark 8 Project Team✅
    |---[4/8] Sales and Marketing✅
    |---[5/8] Sample Team Site✅
    |---[6/8] Digital Initiative Public Relations✅
    |---[7/8] p70m✅
    |---[8/8] U.S. Sales✅
  ```

Part of New.ps1 output:

```powershell
[3/137] Extracting [AADGroup]...
    |---[1/9] All Company✅
    |---[2/9] Retail✅
    |---[3/9] Mark 8 Project Team✅
    |---[4/9] Sales and Marketing✅
    |---[5/9] Sample Team Site✅
    |---[6/9] Digital Initiative Public Relations✅
    |---[7/9] p70m✅
    |---[8/9] Henken✅
    |---[9/9] U.S. Sales✅
```

In the New.ps1 I can definately see the group:
```powershell
        AADGroup 6aff0c77-0e37-4eec-9d91-734e0f8f5113
        {
            Credential           = $Credscredential;
            DisplayName          = "Henken";
            Ensure               = "Present";
            GroupTypes           = @();
            Id                   = "ba34e3cd-c071-4895-b13d-7845cbcd933f";
            MailEnabled          = $False;
            MailNickname         = "4398bea7-9";
            SecurityEnabled      = $True;
        }
```

# Expected behavior

I'd expected a HTML report showing the configuration policy (which I don't seem to find in the output nor the `New.ps1`) and the AAD Group `Henken`.

**Comment by AbdelmounaimZaidi** [comment]
Having the same issue here. Any feedback would be appreciated.

**Comment by ykuijs** [comment]
I have been testing the behavior. I am not sure where that Settings error is coming from.

Can you please share the two configs you are trying compare?

FYI: The Delta parameter can be used to generate a report from two already compared results. It isn't a boolean value. So the second example you shared is the expected results. For more info about the Delta parameter, see this article: https://microsoft365dsc.com/user-guide/cmdlets/New-M365DSCDeltaReport/

**Comment by krugerd** [comment]
I have noticed a related issue with New-m365dscdeltareport.  The DeltaReport.html output report shows under "Resources that are configured differently" that the values for (for example) TargetedDomainsToProtect are different.  If we take a look at what the actual values are, we see:

C:\ps\DSC\delta>type source.ps1 | find "TargetedDomainsToProtect"
            TargetedDomainsToProtect                      = @();
            TargetedDomainsToProtect                      = @();

C:\ps\DSC\delta>type delta.ps1 | find "TargetedDomainsToProtect"

C:\ps\DSC\delta>type delta.ps1

So they are reported as different.  is this the expected behaviour?  On the report it looks like a bug as both values show up as "empty".

Thanks!

**Comment by teriaavibes** [comment]
Similar issue here 
`Error: From
Error: From
Error: Settings
Cannot index into a null array.
At C:\Program Files\WindowsPowerShell\Modules\Microsoft365DSC\1.22.608.1\modules\M365DSCReport.psm1:633 char:72
+ ... terScript { $_.ResourceName -eq $currentDestinationResource.ResourceN ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
    + FullyQualifiedErrorId : NullArray

Cannot index into a null array.
At C:\Program Files\WindowsPowerShell\Modules\Microsoft365DSC\1.22.608.1\modules\M365DSCReport.psm1:634 char:9
+         $currentDestinationKeyValue = $currentDestinationResource.($k ...
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
    + FullyQualifiedErrorId : NullArray

Cannot index into a null array.
At C:\Program Files\WindowsPowerShell\Modules\Microsoft365DSC\1.22.608.1\modules\M365DSCReport.psm1:633 char:72
+ ... terScript { $_.ResourceName -eq $currentDestinationResource.ResourceN ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
    + FullyQualifiedErrorId : NullArray

Cannot index into a null array.
At C:\Program Files\WindowsPowerShell\Modules\Microsoft365DSC\1.22.608.1\modules\M365DSCReport.psm1:634 char:9
+         $currentDestinationKeyValue = $currentDestinationResource.($k ...
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
    + FullyQualifiedErrorId : NullArray`'
    
And the html report looks the same, no information in it.

**Comment by andikrueger** [comment]
@teriaavibes Could you provide a configuration to reproduce this issue?

**Comment by GarySCTS** [comment]
Same issue here with version 1.22.615.1. Do an export Export-M365DSCConfiguration, copy file so it's identical. Run New-M365DSCDeltaReport. It generates 5 mismatches on the report. I manually checked these reports just to be sure. The yare identical as expected.

**Comment by andikrueger** [comment]
There is a new issue to track the breaking change for SCSensitivityLabel #2314

**Comment by ykuijs** [comment]
This issue should be fixed with the 1.22.1005.1 release. Can you please test and confirm?

**Comment by NikCharlebois** [comment]
As mentioned by @ykuijs, this would have been fixed with the Fall Major release. We can re-open later if the issue still persists for you.
